### PR TITLE
katsu update

### PR DESCRIPTION
**Pull request by SonQBChau**
- bump katsu version to 2.0.0
- update katsu_service to part_31
- can update to python 3.11 and alpine 3.17 by changing .env VENV_PYTHON and ALPINE_VERSION

How to test:
- delete katsu and db container, images and volumes (make version changes in .env if you want)
- go to cd/katsu/katsu_service and checkout [sonchau/part_31](https://github.com/CanDIG/katsu/pull/114)
- make build-katsu and make compose-katsu
- make test-integration, all should pass

### CODE REVIEW

File: lib/katsu/docker-compose.yml
Comment (daisieh) [inline lib/katsu/docker-compose.yml]: These values should be picked up from the .env file, like so:
```
        venv_python: "${VENV_PYTHON}"
        alpine_version: "${ALPINE_VERSION}"
```

File: lib/katsu/docker-compose.yml
Comment (daisieh) [inline lib/katsu/docker-compose.yml]: Are you sure we should bump this here? Every other docker-compose uses 3.7. If we bump this, we should bump it everywhere.

File: lib/katsu/docker-compose.yml
Comment (SonQBChau) [reply lib/katsu/docker-compose.yml]: ya, this makes sense. The old docker-compose didn't use env vars so I didn't know, but this would bump the version for all others. Do we need to wait until they are tested out?

File: lib/katsu/docker-compose.yml
Comment (SonQBChau) [reply lib/katsu/docker-compose.yml]: again, didn't know this, can leave 3.7 if we don't want to bump everywhere

File: lib/katsu/docker-compose.yml
Comment (SonQBChau) [reply lib/katsu/docker-compose.yml]: I did a quick test and federation failed to compose:
```
IGv2/lib/federation-service/docker-compose.yml build 
[+] Building 13.9s (7/11)                                                                                                                               
 => [internal] load build definition from Dockerfile                                                                                               0.0s
 => => transferring dockerfile: 32B                                                                                                                0.0s
 => [internal] load .dockerignore                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/python:3.11-alpine3.17                                                                          0.3s
 => [1/7] FROM docker.io/library/python:3.11-alpine3.17@sha256:507818d46649f8543e58d19a00e3a1a428bb7e87c0bf7f7d1ffe7b076cda11be                    0.0s
 => [internal] load build context                                                                                                                  0.0s
 => => transferring context: 9.24kB                                                                                                                0.0s
 => CACHED [2/7] RUN apk update                                                                                                                    0.0s
 => ERROR [3/7] RUN apk add --no-cache   autoconf   automake   make   gcc   linux-headers   perl   bash   build-base   musl-dev   zlib-dev   bzi  13.5s
------                                                                                                                                                  
 > [3/7] RUN apk add --no-cache   autoconf   automake   make   gcc   linux-headers   perl   bash   build-base   musl-dev   zlib-dev   bzip2-dev   xz-dev   libcurl   curl   curl-dev   yaml-dev   libressl-dev   pcre-dev   git:
#0 0.143 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/main/aarch64/APKINDEX.tar.gz
#0 0.565 fetch https://dl-cdn.alpinelinux.org/alpine/v3.17/community/aarch64/APKINDEX.tar.gz
#0 0.954 (1/57) Upgrading libcrypto3 (3.0.8-r3 -> 3.0.8-r4)
#0 1.069 (2/57) Upgrading libssl3 (3.0.8-r3 -> 3.0.8-r4)
#0 1.113 (3/57) Installing m4 (1.4.19-r1)
#0 1.142 (4/57) Installing perl (5.36.0-r1)
#0 1.682 (5/57) Installing autoconf (2.71-r1)
#0 1.745 (6/57) Installing automake (1.16.5-r1)
#0 1.801 (7/57) Installing bash (5.2.15-r0)
#0 1.856 Executing bash-5.2.15-r0.post-install
#0 1.859 (8/57) Installing libgcc (12.2.1_git20220924-r4)
#0 1.894 (9/57) Installing libstdc++ (12.2.1_git20220924-r4)
#0 1.979 (10/57) Installing binutils (2.39-r2)
#0 2.199 (11/57) Installing libmagic (5.43-r0)
#0 2.269 (12/57) Installing file (5.43-r0)
#0 2.295 (13/57) Installing libgomp (12.2.1_git20220924-r4)
#0 2.329 (14/57) Installing libatomic (12.2.1_git20220924-r4)
#0 2.355 (15/57) Installing gmp (6.2.1-r2)
#0 2.392 (16/57) Installing isl25 (0.25-r1)
#0 2.474 (17/57) Installing mpfr4 (4.1.0-r0)
#0 2.511 (18/57) Installing mpc1 (1.2.1-r1)
#0 2.538 (19/57) Installing gcc (12.2.1_git20220924-r4)
#0 6.040 (20/57) Installing libstdc++-dev (12.2.1_git20220924-r4)
#0 7.608 (21/57) Installing musl-dev (1.2.3-r4)
#0 7.986 (22/57) Installing libc-dev (0.7.2-r3)
#0 8.011 (23/57) Installing g++ (12.2.1_git20220924-r4)
#0 9.732 (24/57) Installing make (4.3-r1)
#0 9.774 (25/57) Installing fortify-headers (1.1-r1)
#0 9.800 (26/57) Installing patch (2.7.6-r9)
#0 9.850 (27/57) Installing build-base (0.5-r3)
#0 9.875 (28/57) Installing pkgconf (1.9.4-r0)
#0 9.913 (29/57) Installing bzip2-dev (1.0.8-r4)
#0 9.936 (30/57) Installing brotli-libs (1.0.9-r9)
#0 10.00 (31/57) Installing nghttp2-libs (1.51.0-r0)
#0 10.04 (32/57) Installing libcurl (8.0.1-r0)
#0 10.09 (33/57) Installing curl (8.0.1-r0)
#0 10.14 (34/57) Installing openssl-dev (3.0.8-r4)
#0 10.22 (35/57) Installing nghttp2-dev (1.51.0-r0)
#0 10.25 (36/57) Installing zlib-dev (1.2.13-r0)
#0 10.29 (37/57) Installing brotli (1.0.9-r9)
#0 10.32 (38/57) Installing brotli-dev (1.0.9-r9)
#0 10.35 (39/57) Installing curl-dev (8.0.1-r0)
#0 10.38 (40/57) Installing pcre2 (10.42-r0)
#0 10.43 (41/57) Installing git (2.38.5-r0)
#0 10.86 (42/57) Installing perl-error (0.17029-r1)
#0 10.89 (43/57) Installing perl-git (2.38.5-r0)
#0 10.93 (44/57) Installing git-perl (2.38.5-r0)
#0 10.95 (45/57) Installing libressl3.6-libcrypto (3.6.2-r0)
#0 11.09 (46/57) Installing libressl3.6-libssl (3.6.2-r0)
#0 11.13 (47/57) Installing libressl3.6-libtls (3.6.2-r0)
#0 11.27 (48/57) Installing libressl-dev (3.6.2-r0)
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/aes.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/asn1.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/asn1t.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/bio.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/blowfish.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/bn.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/buffer.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/camellia.h owned by openssl-dev-3.0.8-r4.
#0 11.31 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/cast.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/cmac.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/cms.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/comp.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/conf.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/conf_api.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/crypto.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ct.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/des.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/dh.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/dsa.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/dtls1.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ec.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ecdh.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ecdsa.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/engine.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/err.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/evp.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/hmac.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/idea.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/kdf.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/lhash.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/md4.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/md5.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/modes.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/obj_mac.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/objects.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ocsp.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/opensslconf.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/opensslv.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ossl_typ.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/pem.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/pem2.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/pkcs12.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/pkcs7.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/rand.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/rc2.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/rc4.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ripemd.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/rsa.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/safestack.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/sha.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/srtp.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ssl.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ssl2.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ssl3.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/stack.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/tls1.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ts.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/txt_db.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/ui.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/whrlpool.h owned by openssl-dev-3.0.8-r4.
#0 11.32 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/x509.h owned by openssl-dev-3.0.8-r4.
#0 11.33 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/x509_vfy.h owned by openssl-dev-3.0.8-r4.
#0 11.33 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/include/openssl/x509v3.h owned by openssl-dev-3.0.8-r4.
#0 11.91 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/lib/libcrypto.so owned by openssl-dev-3.0.8-r4.
#0 12.05 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/lib/libssl.so owned by openssl-dev-3.0.8-r4.
#0 12.77 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/lib/pkgconfig/libcrypto.pc owned by openssl-dev-3.0.8-r4.
#0 12.77 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/lib/pkgconfig/libssl.pc owned by openssl-dev-3.0.8-r4.
#0 12.77 ERROR: libressl-dev-3.6.2-r0: trying to overwrite usr/lib/pkgconfig/openssl.pc owned by openssl-dev-3.0.8-r4.
#0 12.77 (49/57) Installing linux-headers (5.19.5-r0)
#0 12.99 (50/57) Installing libpcre16 (8.45-r2)
#0 13.03 (51/57) Installing libpcre32 (8.45-r2)
#0 13.07 (52/57) Installing pcre (8.45-r2)
#0 13.10 (53/57) Installing libpcrecpp (8.45-r2)
#0 13.13 (54/57) Installing pcre-dev (8.45-r2)
#0 13.22 (55/57) Installing xz-dev (5.2.9-r0)
#0 13.25 (56/57) Installing yaml (0.2.5-r0)
#0 13.28 (57/57) Installing yaml-dev (0.2.5-r0)
#0 13.31 Executing busybox-1.35.0-r29.trigger
#0 13.32 Executing ca-certificates-20220614-r4.trigger
#0 13.36 1 error; 362 MiB in 92 packages
------
failed to solve: executor failed running [/bin/sh -c apk add --no-cache   autoconf   automake   make   gcc   linux-headers   perl   bash   build-base   musl-dev   zlib-dev   bzip2-dev   xz-dev   libcurl   curl   curl-dev   yaml-dev   libressl-dev   pcre-dev   git]: exit code: 1
make[2]: *** [build-federation-service] Error 17
make[1]: *** [build-images] Error 2
make: *** [build-all] Error 2
```

We can either wait until all the repos get fixed and merge or go with this version and change later?

Comment (daisieh) [reply lib/katsu/docker-compose.yml]: I think we should change it here to the env vars and worry about updating them all later.

**Review by daisieh — APPROVED** [review]
Works well as instructed.

**Pull request merged**
